src/main/java/org/palladiosimulator/dependencytool/EmbeddedNeo4j.java

```diff
@@ -0,0 +1,180 @@
+package org.palladiosimulator.dependencytool;
+
+import static org.neo4j.configuration.GraphDatabaseSettings.DEFAULT_DATABASE_NAME;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.neo4j.dbms.api.DatabaseManagementService;
+import org.neo4j.dbms.api.DatabaseManagementServiceBuilder;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.io.fs.FileUtils;
+
+public class EmbeddedNeo4j {
+    private enum Relationships implements RelationshipType {
+        REPOSITORY("depending on repository"), FEATURE("require the feature"), BUNDLE("require the bundle");
+
+        private final String message;
+
+        Relationships(String message) {
+            this.message = message;
+        }
+
+        public String getMessage() {
+            return this.message;
+        }
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(EmbeddedNeo4j.class.getName());
+
+    private static void registerShutdownHook(final DatabaseManagementService managementService) {
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                managementService.shutdown();
+            }
+        });
+    }
+
+    private final Path databaseDirectory;
+    private final GraphDatabaseService databaseService;
+    private final DatabaseManagementService managementService;
+
+    private final Label repositoryLabel = Label.label("Repository");
+    private final Label featureLabel = Label.label("Feature");
+    private final Label bundleLabel = Label.label("Bundle");
+
+    private final Map<String, Long> repositoryIds = new HashMap<>();
+    private final Map<String, Long> featureIds = new HashMap<>();
+    private final Map<String, Long> bundleIds = new HashMap<>();
+
+    public EmbeddedNeo4j() {
+        this.databaseDirectory = Path.of(DEFAULT_DATABASE_NAME).toAbsolutePath().normalize();
+
+        try {
+            LOGGER.info("Delete Directory: " + this.databaseDirectory);
+            FileUtils.deleteDirectory(this.databaseDirectory);
```

Comment:
The name of the directory is not mentioned anywhere (for users).  However, this unknown directory is deleted. Depending on the current working directory, this might lead to data loss. I suggest checking if the directory exists and stop execution with a warning.